MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -709,6 +722,121 @@ std::string EnggDiffFittingPresenter::functionStrFactory(
   return functionStr;
 }
 
+void EnggDiffFittingPresenter::browsePeaksToFit() {
+  try {
+    QString prevPath = QString::fromStdString(m_view->focusingDir());
+    if (prevPath.isEmpty()) {
+      prevPath = QString::fromStdString(m_view->getPreviousDir());
+    }
+
+    std::string path = m_view->getOpenFile(prevPath.toStdString());
+
+    if (path.empty()) {
+      return;
+    }
+
+    m_view->setPreviousDir(path);
+
+    std::string peaksData = readPeaksFile(path);
+
+    m_view->setPeakList(peaksData);
+  } catch (...) {
+    m_view->userWarning(
+        "Unable to import the peaks from a file: ",
+        "File corrupted or could not be opened. Please try again");
+    return;
+  }
+}
+
+void EnggDiffFittingPresenter::addPeakToList() {
+
+  if (m_view->peakPickerEnabled()) {
+    auto peakCentre = m_view->getPeakCentre();
+
+    std::stringstream stream;
+    stream << std::fixed << std::setprecision(4) << peakCentre;
+    auto strPeakCentre = stream.str();
+
+    QString curExpPeaksList =
+        QString::fromStdString(m_view->fittingPeaksData());
+    QString comma = ",";
```

Comment:
No need to convert to `QString`, here all the operations can be done with `std::string`.